File: 1101-1200/1166-Design File System/1166-Design File System.cpp
class FileSystem {
public:
    FileSystem() {
        table[""] = -1;
    }
    
    bool createPath(string path, int value) {
        string parent = path.substr(0, path.rfind("/"));
        if (!table.count(parent) || table.count(path)) {
            return false;
        }
        else {
            table[path] = value;
            return true;
        }
    }
    
    int get(string path) {
        return table.count(path) ? table[path] : -1;
    }
    
private:
    unordered_map<string, int> table;
};

/**
 * Your FileSystem object will be instantiated and called as such:
 * FileSystem* obj = new FileSystem();
 * bool param_1 = obj->createPath(path,value);
 * int param_2 = obj->get(path);
 */

File: 1101-1200/1166-Design File System/1166-Design File System.py
class FileSystem:

    def __init__(self):
        self.table = {'': -1}

    def createPath(self, path: str, value: int) -> bool:
        parent = path[:path.rfind('/')]
        if parent not in self.table or path in self.table:
            return False
        else:
            self.table[path] = value
            return True

    def get(self, path: str) -> int:
        return self.table.get(path, -1)


# Your FileSystem object will be instantiated and called as such:
# obj = FileSystem()
# param_1 = obj.createPath(path,value)
# param_2 = obj.get(path)
